src/core/jwk.rs

```diff
@@ -315,28 +398,32 @@ impl
 const RSA_HEADER: &str = "-----BEGIN RSA PRIVATE KEY-----";
 const RSA_FOOTER: &str = "-----END RSA PRIVATE KEY-----";
 
+pub trait RngClone: dyn_clone::DynClone + rand::RngCore + rand::CryptoRng {}
```

Comment:
Yeah it's not great... I added a comment to explain why it's so intricate. But I think if the `sign` function started accepting `&mut self` it could be simplified?

As for the security concern, I'm currently confused whether it's actually cloning the RNG or just the Box because the tests kept working apart from the PSS ones

Reply:
That's fair enough 👍 Just be aware that the `dyn_clone` crate is full of unsafe code. But it's a very small library and trustworthy.

Reply:
anything written by @dtolnay with 11M+ downloads sounds good to me :-)